include/memcache_functions.php

```diff
@@ -23,3 +23,43 @@ function cdash_memcache_key($page_name)
                               $page_name,
                               md5(serialize($_REQUEST))));
 }
+
+function cdash_memcache_connect($server, $port)
+{
+    if (class_exists('Memcached')) {
+        $memcache = new Memcached();
+        $memcache->setOption(Memcached::OPT_COMPRESSION, true);
+        global $CDASH_USE_ELASTICACHE_AUTO_DISCOVERY;
+        if ($CDASH_USE_ELASTICACHE_AUTO_DISCOVERY) {
+            $memcache->setOption(Memcached::OPT_CLIENT_MODE, Memcached::DYNAMIC_CLIENT_MODE);
+        }
+        if ($memcache->addServer($server, $port) !== false) {
+            return $memcache;
+        }
+    } elseif (class_exists('Memcache')) {
+        $memcache = new Memcache();
+        if ($memcache->connect($server, $port) !== false) {
+            return $memcache;
+        }
+    }
+    return false;
+}
+
+function cdash_memcache_get($memcache, $key)
+{
+    if ($memcache === false) {
+        return false;
+    }
+    return $memcache->get($key);
+}
+
+function cdash_memcache_set($memcache, $key, $var, $expire)
+{
+    if ($memcache instanceof Memcached) {
+        return $memcache->set($key, $var, $expire);
+    }
+    if ($memcache instanceof Memcache) {
+        return $memcache->set($key, $var, MEMCACHE_COMPRESSED, $expire);
+    }
```

Comment:
Is there a reason we have to support Memcache at the moment?

If not, should we just scrap all Memcache related code and integrate doctrine/cache when we have to support something else? Just looking to avoid `if memcached .. else if memcache .. elseif thenextthing` if possible.